internal/util/tls/tls_cert.go

```diff
@@ -7,8 +7,9 @@ import (
 )
 
 // ExtractClientCertificates extracts tls.Certificates from TLSClientConfig.
-// It returns an empty slice in case there was no client cert and/or client key provided.
-func ExtractClientCertificates(cert []byte, certFile string, key []byte, keyFile string) ([]tls.Certificate, error) {
+// It returns nil in case there was no client cert and/or client key provided.
+// REVIEW: in case of no certs specified, return nil, nil, OR return non-nil error, OR add a boolean return value?
```

Comment:
The version you proposed works fine for me - as long as we have it explicitly stated in the comment that `nil, nil` means there was no certificate, it's ok.

There's one more option - going with monads 😅 being explicit about the value optionality: https://go.dev/play/p/vQOfnwIAPWg

Reply:
I'm fine with both. But I feel that there's no need for that in here. Unless you convince me.

Rationale and/or benchmark might work ;)

Reply:
A rationale for using mo.Optional? I believe that would be being explicit that the value is optional by expressing it directly in the returned type. `mo.Optional[tls.Certificate]` doesn't need an additional comment about the `nil` being a valid value which means there was no certificate. 

But yeah, I brought it up just to signal that we might consider that, I don't feel like there's a real need here, and expressing optionality by making a returned value a pointer is kinda convention already.

Reply:
> But yeah, I brought it up just to signal that we might consider that, I don't feel like there's a real need here, and expressing optionality by making a returned value a pointer is kinda convention already.

Alright. Let's leave it at using the pointer for now (in this PR) but I'm happy to revisit that in a dedicated PR 👍